// @acuvity/acuvity/dist/esm/lib/config.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { pathToFunc } from "./url.js";
/**
 * Contains the list of servers available to the SDK
 */
export const ServerList = [
    /**
     * The Apex API server which is specific to every organization.
     *
     * @remarks
     * The apex_domain and apex_port variables can be determined by a call to
     * the well-known Apex info endpoint on the backend.
     */
    "https://{apex_domain}:{apex_port}",
];
export function serverURLFromOptions(options) {
    let serverURL = options.serverURL;
    const serverParams = [
        {
            "apex_domain": options.apexDomain ?? "apex.acuvity.ai",
            "apex_port": options.apexPort ?? "443",
        },
    ];
    let params = {};
    if (!serverURL) {
        const serverIdx = options.serverIdx ?? 0;
        if (serverIdx < 0 || serverIdx >= ServerList.length) {
            throw new Error(`Invalid server index ${serverIdx}`);
        }
        serverURL = ServerList[serverIdx] || "";
        params = serverParams[serverIdx] || {};
    }
    const u = pathToFunc(serverURL)(params);
    return new URL(u);
}
export const SDK_METADATA = {
    language: "typescript",
    openapiDocVersion: "1.0",
    sdkVersion: "0.4.0",
    genVersion: "2.488.4",
    userAgent: "speakeasy-sdk/typescript 0.4.0 2.488.4 1.0 @acuvity/acuvity",
};
