// @acuvity/acuvity/dist/esm/models/components/latency.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const Latency$inboundSchema = z.object({
    accessPolicy: z.number().int().optional(),
    analysis: z.number().int().optional(),
    assignPolicy: z.number().int().optional(),
    contentPolicy: z.number().int().optional(),
    extraction: z.number().int().optional(),
});
/** @internal */
export const Latency$outboundSchema = z.object({
    accessPolicy: z.number().int().optional(),
    analysis: z.number().int().optional(),
    assignPolicy: z.number().int().optional(),
    contentPolicy: z.number().int().optional(),
    extraction: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Latency$;
(function (Latency$) {
    /** @deprecated use `Latency$inboundSchema` instead. */
    Latency$.inboundSchema = Latency$inboundSchema;
    /** @deprecated use `Latency$outboundSchema` instead. */
    Latency$.outboundSchema = Latency$outboundSchema;
})(Latency$ || (Latency$ = {}));
export function latencyToJSON(latency) {
    return JSON.stringify(Latency$outboundSchema.parse(latency));
}
export function latencyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Latency$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Latency' from JSON`);
}
