// @acuvity/acuvity/dist/commonjs/models/components/analyzer.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Analyzer$ = exports.Analyzer$outboundSchema = exports.Analyzer$inboundSchema = void 0;
exports.analyzerToJSON = analyzerToJSON;
exports.analyzerFromJSON = analyzerFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const analyzermodel_js_1 = require("./analyzermodel.js");
const detector_js_1 = require("./detector.js");
/** @internal */
exports.Analyzer$inboundSchema = z.object({
    ID: z.string().optional(),
    description: z.string().optional(),
    detectors: z.array(detector_js_1.Detector$inboundSchema).optional(),
    enabled: z.boolean().optional(),
    group: z.string().optional(),
    models: z.array(analyzermodel_js_1.Analyzermodel$inboundSchema).optional(),
    name: z.string().optional(),
    namespace: z.string().optional(),
    triggers: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ID": "id",
    });
});
/** @internal */
exports.Analyzer$outboundSchema = z.object({
    id: z.string().optional(),
    description: z.string().optional(),
    detectors: z.array(detector_js_1.Detector$outboundSchema).optional(),
    enabled: z.boolean().optional(),
    group: z.string().optional(),
    models: z.array(analyzermodel_js_1.Analyzermodel$outboundSchema).optional(),
    name: z.string().optional(),
    namespace: z.string().optional(),
    triggers: z.array(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        id: "ID",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Analyzer$;
(function (Analyzer$) {
    /** @deprecated use `Analyzer$inboundSchema` instead. */
    Analyzer$.inboundSchema = exports.Analyzer$inboundSchema;
    /** @deprecated use `Analyzer$outboundSchema` instead. */
    Analyzer$.outboundSchema = exports.Analyzer$outboundSchema;
})(Analyzer$ || (exports.Analyzer$ = Analyzer$ = {}));
function analyzerToJSON(analyzer) {
    return JSON.stringify(exports.Analyzer$outboundSchema.parse(analyzer));
}
function analyzerFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Analyzer$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Analyzer' from JSON`);
}
