// @acuvity/acuvity/dist/commonjs/models/components/detector.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Detector$ = exports.Detector$outboundSchema = exports.Detector$inboundSchema = void 0;
exports.detectorToJSON = detectorToJSON;
exports.detectorFromJSON = detectorFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.Detector$inboundSchema = z.object({
    description: z.string().optional(),
    group: z.string().optional(),
    label: z.string().optional(),
    name: z.string().optional(),
    positional: z.boolean().optional(),
});
/** @internal */
exports.Detector$outboundSchema = z.object({
    description: z.string().optional(),
    group: z.string().optional(),
    label: z.string().optional(),
    name: z.string().optional(),
    positional: z.boolean().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Detector$;
(function (Detector$) {
    /** @deprecated use `Detector$inboundSchema` instead. */
    Detector$.inboundSchema = exports.Detector$inboundSchema;
    /** @deprecated use `Detector$outboundSchema` instead. */
    Detector$.outboundSchema = exports.Detector$outboundSchema;
})(Detector$ || (exports.Detector$ = Detector$ = {}));
function detectorToJSON(detector) {
    return JSON.stringify(exports.Detector$outboundSchema.parse(detector));
}
function detectorFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Detector$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Detector' from JSON`);
}
