// @acuvity/acuvity/dist/commonjs/models/components/extraction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Extraction$ = exports.Extraction$outboundSchema = exports.Extraction$inboundSchema = void 0;
exports.extractionToJSON = extractionToJSON;
exports.extractionFromJSON = extractionFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const modality_js_1 = require("./modality.js");
const textualdetection_js_1 = require("./textualdetection.js");
/** @internal */
exports.Extraction$inboundSchema = z.object({
    PIIs: z.record(z.number()).optional(),
    annotations: z.record(z.string()).optional(),
    categories: z.array(modality_js_1.Modality$inboundSchema).optional(),
    confidentiality: z.number().optional(),
    data: z.string().optional(),
    detections: z.array(textualdetection_js_1.Textualdetection$inboundSchema).optional(),
    exploits: z.record(z.number()).optional(),
    hash: z.string().optional(),
    intent: z.record(z.number()).optional(),
    internal: z.boolean().optional(),
    keywords: z.record(z.number()).optional(),
    label: z.string().optional(),
    languages: z.record(z.number()).optional(),
    modalities: z.array(modality_js_1.Modality$inboundSchema).optional(),
    relevance: z.number().optional(),
    secrets: z.record(z.number()).optional(),
    topics: z.record(z.number()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "PIIs": "piIs",
    });
});
/** @internal */
exports.Extraction$outboundSchema = z.object({
    piIs: z.record(z.number()).optional(),
    annotations: z.record(z.string()).optional(),
    categories: z.array(modality_js_1.Modality$outboundSchema).optional(),
    confidentiality: z.number().optional(),
    data: z.string().optional(),
    detections: z.array(textualdetection_js_1.Textualdetection$outboundSchema).optional(),
    exploits: z.record(z.number()).optional(),
    hash: z.string().optional(),
    intent: z.record(z.number()).optional(),
    internal: z.boolean().optional(),
    keywords: z.record(z.number()).optional(),
    label: z.string().optional(),
    languages: z.record(z.number()).optional(),
    modalities: z.array(modality_js_1.Modality$outboundSchema).optional(),
    relevance: z.number().optional(),
    secrets: z.record(z.number()).optional(),
    topics: z.record(z.number()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        piIs: "PIIs",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Extraction$;
(function (Extraction$) {
    /** @deprecated use `Extraction$inboundSchema` instead. */
    Extraction$.inboundSchema = exports.Extraction$inboundSchema;
    /** @deprecated use `Extraction$outboundSchema` instead. */
    Extraction$.outboundSchema = exports.Extraction$outboundSchema;
})(Extraction$ || (exports.Extraction$ = Extraction$ = {}));
function extractionToJSON(extraction) {
    return JSON.stringify(exports.Extraction$outboundSchema.parse(extraction));
}
function extractionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Extraction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Extraction' from JSON`);
}
