// @acuvity/acuvity/dist/commonjs/models/components/latency.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Latency$ = exports.Latency$outboundSchema = exports.Latency$inboundSchema = void 0;
exports.latencyToJSON = latencyToJSON;
exports.latencyFromJSON = latencyFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.Latency$inboundSchema = z.object({
    accessPolicy: z.number().int().optional(),
    analysis: z.number().int().optional(),
    assignPolicy: z.number().int().optional(),
    contentPolicy: z.number().int().optional(),
    extraction: z.number().int().optional(),
});
/** @internal */
exports.Latency$outboundSchema = z.object({
    accessPolicy: z.number().int().optional(),
    analysis: z.number().int().optional(),
    assignPolicy: z.number().int().optional(),
    contentPolicy: z.number().int().optional(),
    extraction: z.number().int().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Latency$;
(function (Latency$) {
    /** @deprecated use `Latency$inboundSchema` instead. */
    Latency$.inboundSchema = exports.Latency$inboundSchema;
    /** @deprecated use `Latency$outboundSchema` instead. */
    Latency$.outboundSchema = exports.Latency$outboundSchema;
})(Latency$ || (exports.Latency$ = Latency$ = {}));
function latencyToJSON(latency) {
    return JSON.stringify(exports.Latency$outboundSchema.parse(latency));
}
function latencyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Latency$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Latency' from JSON`);
}
