// @acuvity/acuvity/dist/commonjs/models/components/policerequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Policerequest$ = exports.Policerequest$outboundSchema = exports.Policerequest$inboundSchema = exports.Type$ = exports.Type$outboundSchema = exports.Type$inboundSchema = exports.Anonymization$ = exports.Anonymization$outboundSchema = exports.Anonymization$inboundSchema = exports.Type = exports.Anonymization = void 0;
exports.policerequestToJSON = policerequestToJSON;
exports.policerequestFromJSON = policerequestFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const extractionrequest_js_1 = require("./extractionrequest.js");
const policeexternaluser_js_1 = require("./policeexternaluser.js");
/**
 * How to anonymize the data. If deanonymize is true, then VariablSize is required.
 */
exports.Anonymization = {
    FixedSize: "FixedSize",
    VariableSize: "VariableSize",
};
/**
 * The type of text.
 */
exports.Type = {
    Input: "Input",
    Output: "Output",
};
/** @internal */
exports.Anonymization$inboundSchema = z.nativeEnum(exports.Anonymization);
/** @internal */
exports.Anonymization$outboundSchema = exports.Anonymization$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Anonymization$;
(function (Anonymization$) {
    /** @deprecated use `Anonymization$inboundSchema` instead. */
    Anonymization$.inboundSchema = exports.Anonymization$inboundSchema;
    /** @deprecated use `Anonymization$outboundSchema` instead. */
    Anonymization$.outboundSchema = exports.Anonymization$outboundSchema;
})(Anonymization$ || (exports.Anonymization$ = Anonymization$ = {}));
/** @internal */
exports.Type$inboundSchema = z.nativeEnum(exports.Type);
/** @internal */
exports.Type$outboundSchema = exports.Type$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Type$;
(function (Type$) {
    /** @deprecated use `Type$inboundSchema` instead. */
    Type$.inboundSchema = exports.Type$inboundSchema;
    /** @deprecated use `Type$outboundSchema` instead. */
    Type$.outboundSchema = exports.Type$outboundSchema;
})(Type$ || (exports.Type$ = Type$ = {}));
/** @internal */
exports.Policerequest$inboundSchema = z.object({
    annotations: z.record(z.string()).optional(),
    anonymization: exports.Anonymization$inboundSchema.default("FixedSize"),
    bypassHash: z.string().optional(),
    extractions: z.array(extractionrequest_js_1.Extractionrequest$inboundSchema).optional(),
    messages: z.array(z.string()).optional(),
    provider: z.string().optional(),
    type: exports.Type$inboundSchema.optional(),
    user: policeexternaluser_js_1.Policeexternaluser$inboundSchema.optional(),
});
/** @internal */
exports.Policerequest$outboundSchema = z.object({
    annotations: z.record(z.string()).optional(),
    anonymization: exports.Anonymization$outboundSchema.default("FixedSize"),
    bypassHash: z.string().optional(),
    extractions: z.array(extractionrequest_js_1.Extractionrequest$outboundSchema).optional(),
    messages: z.array(z.string()).optional(),
    provider: z.string().optional(),
    type: exports.Type$outboundSchema.optional(),
    user: policeexternaluser_js_1.Policeexternaluser$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Policerequest$;
(function (Policerequest$) {
    /** @deprecated use `Policerequest$inboundSchema` instead. */
    Policerequest$.inboundSchema = exports.Policerequest$inboundSchema;
    /** @deprecated use `Policerequest$outboundSchema` instead. */
    Policerequest$.outboundSchema = exports.Policerequest$outboundSchema;
})(Policerequest$ || (exports.Policerequest$ = Policerequest$ = {}));
function policerequestToJSON(policerequest) {
    return JSON.stringify(exports.Policerequest$outboundSchema.parse(policerequest));
}
function policerequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Policerequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Policerequest' from JSON`);
}
