// @acuvity/acuvity/dist/commonjs/models/components/policeexternaluser.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Policeexternaluser$ = exports.Policeexternaluser$outboundSchema = exports.Policeexternaluser$inboundSchema = void 0;
exports.policeexternaluserToJSON = policeexternaluserToJSON;
exports.policeexternaluserFromJSON = policeexternaluserFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.Policeexternaluser$inboundSchema = z.object({
    claims: z.array(z.string()),
    name: z.string(),
});
/** @internal */
exports.Policeexternaluser$outboundSchema = z.object({
    claims: z.array(z.string()),
    name: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Policeexternaluser$;
(function (Policeexternaluser$) {
    /** @deprecated use `Policeexternaluser$inboundSchema` instead. */
    Policeexternaluser$.inboundSchema = exports.Policeexternaluser$inboundSchema;
    /** @deprecated use `Policeexternaluser$outboundSchema` instead. */
    Policeexternaluser$.outboundSchema = exports.Policeexternaluser$outboundSchema;
})(Policeexternaluser$ || (exports.Policeexternaluser$ = Policeexternaluser$ = {}));
function policeexternaluserToJSON(policeexternaluser) {
    return JSON.stringify(exports.Policeexternaluser$outboundSchema.parse(policeexternaluser));
}
function policeexternaluserFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Policeexternaluser$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Policeexternaluser' from JSON`);
}
