// @acuvity/acuvity/dist/commonjs/models/components/principalapp.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Principalapp$ = exports.Principalapp$outboundSchema = exports.Principalapp$inboundSchema = void 0;
exports.principalappToJSON = principalappToJSON;
exports.principalappFromJSON = principalappFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.Principalapp$inboundSchema = z.object({
    labels: z.array(z.string()).optional(),
    name: z.string().optional(),
    tier: z.string().optional(),
});
/** @internal */
exports.Principalapp$outboundSchema = z.object({
    labels: z.array(z.string()).optional(),
    name: z.string().optional(),
    tier: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Principalapp$;
(function (Principalapp$) {
    /** @deprecated use `Principalapp$inboundSchema` instead. */
    Principalapp$.inboundSchema = exports.Principalapp$inboundSchema;
    /** @deprecated use `Principalapp$outboundSchema` instead. */
    Principalapp$.outboundSchema = exports.Principalapp$outboundSchema;
})(Principalapp$ || (exports.Principalapp$ = Principalapp$ = {}));
function principalappToJSON(principalapp) {
    return JSON.stringify(exports.Principalapp$outboundSchema.parse(principalapp));
}
function principalappFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Principalapp$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Principalapp' from JSON`);
}
