// @acuvity/acuvity/dist/commonjs/models/components/textualdetection.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Textualdetection$ = exports.Textualdetection$outboundSchema = exports.Textualdetection$inboundSchema = exports.TextualdetectionType$ = exports.TextualdetectionType$outboundSchema = exports.TextualdetectionType$inboundSchema = exports.TextualdetectionType = void 0;
exports.textualdetectionToJSON = textualdetectionToJSON;
exports.textualdetectionFromJSON = textualdetectionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * The type of detection.
 */
exports.TextualdetectionType = {
    Keyword: "Keyword",
    Pii: "PII",
    Secret: "Secret",
};
/** @internal */
exports.TextualdetectionType$inboundSchema = z.nativeEnum(exports.TextualdetectionType);
/** @internal */
exports.TextualdetectionType$outboundSchema = exports.TextualdetectionType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TextualdetectionType$;
(function (TextualdetectionType$) {
    /** @deprecated use `TextualdetectionType$inboundSchema` instead. */
    TextualdetectionType$.inboundSchema = exports.TextualdetectionType$inboundSchema;
    /** @deprecated use `TextualdetectionType$outboundSchema` instead. */
    TextualdetectionType$.outboundSchema = exports.TextualdetectionType$outboundSchema;
})(TextualdetectionType$ || (exports.TextualdetectionType$ = TextualdetectionType$ = {}));
/** @internal */
exports.Textualdetection$inboundSchema = z.object({
    end: z.number().int().optional(),
    key: z.string().optional(),
    name: z.string().optional(),
    redacted: z.boolean().optional(),
    redactedEnd: z.number().int().optional(),
    redactedStart: z.number().int().optional(),
    score: z.number().optional(),
    start: z.number().int().optional(),
    type: exports.TextualdetectionType$inboundSchema.optional(),
});
/** @internal */
exports.Textualdetection$outboundSchema = z.object({
    end: z.number().int().optional(),
    key: z.string().optional(),
    name: z.string().optional(),
    redacted: z.boolean().optional(),
    redactedEnd: z.number().int().optional(),
    redactedStart: z.number().int().optional(),
    score: z.number().optional(),
    start: z.number().int().optional(),
    type: exports.TextualdetectionType$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Textualdetection$;
(function (Textualdetection$) {
    /** @deprecated use `Textualdetection$inboundSchema` instead. */
    Textualdetection$.inboundSchema = exports.Textualdetection$inboundSchema;
    /** @deprecated use `Textualdetection$outboundSchema` instead. */
    Textualdetection$.outboundSchema = exports.Textualdetection$outboundSchema;
})(Textualdetection$ || (exports.Textualdetection$ = Textualdetection$ = {}));
function textualdetectionToJSON(textualdetection) {
    return JSON.stringify(exports.Textualdetection$outboundSchema.parse(textualdetection));
}
function textualdetectionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Textualdetection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Textualdetection' from JSON`);
}
